sdk/identity/Azure.Identity/src/EnvironmentCredential.cs

```diff
@@ -87,5 +109,59 @@ public override async Task<AccessToken> GetTokenAsync(TokenRequestContext reques
         {
             return (_credential != null) ? await _credential.GetTokenAsync(requestContext, cancellationToken).ConfigureAwait(false) : default;
         }
+
+        (AccessToken, Exception) IExtendedTokenCredential.GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
```

Comment:
I really don't like how complicated the code looks with this tuple. Is it worth adding an internal struct to make the code cleaner?

Reply:
You can also put the 
```
if (ex != null)
throw ex;
```

Logic there

Reply:
Added ExtendedAccessToken struct to carry both AccessToken and the Exception.